Content.Client/UserInterface/Controls/RadialMenu.cs

```diff
@@ -205,51 +278,329 @@ private void OnClicked(ButtonEventArgs args)
+                return menu;
         }
 
         return null;
     }
 }
 
+public interface IRadialMenuItemWithSector
+{
+    /// <summary>
+    /// Angle in radian where button sector should start.
+    /// </summary>
+    public float AngleSectorFrom { set; }
+
+    /// <summary>
+    /// Angle in radian where button sector should end.
+    /// </summary>
+    public float AngleSectorTo { set; }
+
+    /// <summary>
+    /// Outer radius for drawing segment and pointer detection.
+    /// </summary>
+    public float OuterRadius { set; }
+
+    /// <summary>
+    /// Outer radius for drawing segment and pointer detection.
+    /// </summary>
+    public float InnerRadius { set; }
+
+    /// <summary>
+    /// Offset in radian by which menu button should be rotated.
+    /// </summary>
+    public float AngleOffset { set; }
+
+    /// <summary>
+    /// Coordinates of center in parent component - button container.
+    /// </summary>
+    public Vector2 ParentCenter { set; }
+}
+
 [Virtual]
-public class RadialMenuTextureButton : TextureButton
+public class RadialMenuTextureButtonWithSector : RadialMenuTextureButton, IRadialMenuItemWithSector
 {
+    private Vector2[]? _sectorPointsForDrawing;
+
+    private float _angleSectorFrom;
+    private float _angleSectorTo;
+    private float _outerRadius;
+    private float _innerRadius;
+    private float _angleOffset;
+
+    private bool _isWholeCircle;
+    private Vector2? _parentCenter;
+
+    private Color _backgroundColorSrgb = Color.ToSrgb(new Color(70, 73, 102, 128));
+    private Color _hoverBackgroundColorSrgb = Color.ToSrgb(new Color(87, 91, 127, 128));
+    private Color _borderColorSrgb = Color.ToSrgb(new Color(173, 216, 230, 70));
+    private Color _hoverBorderColorSrgb = Color.ToSrgb(new Color(87, 91, 127, 128));
+
     /// <summary>
-    /// Upon clicking this button the radial menu will be moved to the named layer
+    /// Marker, that control should render border of segment. Is false by default.
     /// </summary>
-    public string TargetLayer { get; set; } = string.Empty;
+    /// <remarks>
+    /// By default color of border is same as color of background. Use <see cref="BorderColor"/>
+    /// and <see cref="HoverBorderColor"/> to change it.
+    /// </remarks>
+    public bool DrawBorder { get; set; } = false;
+
+    /// <summary>
+    /// Marker, that control should render background of all sector. Is true by default.
+    /// </summary>
+    public bool DrawBackground { get; set; } = true;
+
+    /// <summary>
+    /// Marker, that control should render separator lines.
+    /// Separator lines are used to visually separate sector of radial menu items.
+    /// Is true by default
+    /// </summary>
+    public bool DrawSeparators { get; set; } = true;
+
+    /// <summary>
+    /// Color of background in non-hovered state. Accepts RGB color, works with sRGB for DrawPrimitive internally.
+    /// </summary>
+    public Color BackgroundColor
```

Comment:
Hmm, this is a bit sussy. It's probably better if we just call this field 'BackgroundColorSrgb' (same for those below) and expect people to supply a srgb formatted value rather than this conversion back and forth. Sorry 'bout misleading you on that

Reply:
There is not 'srgb c-tor', so its just plain harder to supply correct  thing - i see no reason to enfoce developer to do more action that is actually required. Stylesheet code includes only RGB and not sRGB colors as far as i can see at first glance, so if it will be applied - it should break notion just for this control, and also contain more code then it is crucial to declare stylesheet info (transform to srgb). Getter/Setter logic IS bad when you are going to apply it every second /tick or so, but thats clearly not he case, right?

Reply:
OK. Let's leave it for now and see what any other rmaintainer thinks